src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.cs

```diff
@@ -89,7 +89,7 @@ private void SetHttpContext(HttpContext httpContext)
             formData.SetFormData(handler, new FormCollectionReadOnlyDictionary(form));
         }
 
-        if (httpContext.RequestServices.GetService<AntiforgeryStateProvider>() is EndpointAntiforgeryStateProvider antiforgery)
+        if (httpContext.RequestServices.GetService<DefaultAntiforgeryStateProvider>() is EndpointAntiforgeryStateProvider antiforgery)
```

Comment:
Don't we want to look it up as `AntiforgeryStateProvider` rather than `DefaultAntiforgeryStateProvider` here? Or is the intention to ensure people don't register a custom `AntiforgeryStateProvider`?

Reply:
Actually forget this, it doesn't make any difference in this specific case